zhaquirks/tuya/ts0001.py

```diff
@@ -0,0 +1,148 @@
+
+    async def command(
+        self,
+        command_id: Union[foundation.GeneralCommand, int, t.uint8_t],
+        *args,
+        manufacturer: Optional[Union[int, t.uint16_t]] = None,
+        expect_reply: bool = True,
+        tsn: Optional[Union[int, t.uint8_t]] = None,
+        **kwargs: Any,
+    ):
+        """Override the default Cluster command."""
+
+        return await super().command(
+            TUYA_SEND_DATA,
+            *args,
+            manufacturer=foundation.ZCLHeader.NO_MANUFACTURER_ID,
+            expect_reply=expect_reply,
+            tsn=tsn,
+            **kwargs,
+        )
+
+    dp_to_attribute: Dict[int, DPToAttributeMapping] = {
+        1: DPToAttributeMapping(TuyaOnOff.ep_attribute, "on_off"),
+        # Mode
+        101: DPToAttributeMapping(
+            TuyaMCUCluster.ep_attribute,
+            "mode",
+            converter=lambda x: FingerBotMode(x),
+        ),
+        # Down Movement
+        102: DPToAttributeMapping(
+            TuyaMCUCluster.ep_attribute,
+            "down_movement",
+        ),
+        # Sustain Time
+        103: DPToAttributeMapping(TuyaMCUCluster.ep_attribute, "sustain_time"),
+        # Reverse
+        104: DPToAttributeMapping(
+            TuyaMCUCluster.ep_attribute,
+            "reverse",
+            converter=lambda x: FingerBotReverse(x),
+        ),
+        # Battery
+        105: DPToAttributeMapping(TuyaMCUCluster.ep_attribute, "battery"),
+        # Up Movement
+        106: DPToAttributeMapping(
+            TuyaMCUCluster.ep_attribute,
+            "up_movement",
+        ),
+        107: DPToAttributeMapping(TuyaMCUCluster.ep_attribute, "touch_control"),
+    }
+
+    data_point_handlers = {
+        1: "_dp_2_attr_update",
+        101: "_dp_2_attr_update",
+        102: "_dp_2_attr_update",
+        103: "_dp_2_attr_update",
+        104: "_dp_2_attr_update",
+        105: "_dp_2_attr_update",
+        106: "_dp_2_attr_update",
+        107: "_dp_2_attr_update",
+    }
+
+
+class TuyaFingerbot(EnchantedDevice):
+    signature = {
+        MODELS_INFO: [("_TZ3210_dse8ogfy", "TS0001")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    TuyaOnOff.cluster_id,
+                    TuyaFingerbotCluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [
+                    Time.cluster_id,
+                    Ota.cluster_id,
+                ],
+            }
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Groups.cluster_id,
+                    Scenes.cluster_id,
+                    TuyaOnOff.cluster_id,
```

Comment:
The `OnOff` cluster isn't actually replaced by the custom `TuyaOnOff` cluster.
If you want to do that, you have to leave out the `.cluster_id`.

The "Tuya spell" will also work then, since that cluster is bound by ZHA, it extends `TuyaEnchantableCluster` and you're using `EnchantedDevice`.
Z2M seems to at least "cast the Tuya spell" for this device.

Reply:
Z2M have adding the spell to all tuya device but "normal Zigbee" and MCU devices what  i have seen and looks working OK.